### PR TITLE
Pass runPolicy fields to the launcher Job

### PR DESCRIPTION
**Pull request by alculquicondor**
Adds support for:
- ttlSecondsAfterFinish
- activeDeadlineSeconds
- backoffLimit

### CODE REVIEW

**Comment by alculquicondor** [comment]
/hold
for #391

**Comment by alculquicondor** [comment]
cc @ahg-g

**Review by gaocegege — COMMENTED** [review]
/assign @Jeffwan @terrytangyuan

File: v2/pkg/apis/kubeflow/v2beta1/types.go [hunk 1]
```diff
@@ -44,9 +44,8 @@ type MPIJobSpec struct {
 	// +optional
 	SlotsPerWorker *int32 `json:"slotsPerWorker,omitempty"`
 
-	// CleanPodPolicy defines the policy that whether to kill pods after the job completes.
-	// Defaults to None.
-	CleanPodPolicy *common.CleanPodPolicy `json:"cleanPodPolicy,omitempty"`
+	// RunPolicy encapsulates various runtime policies of the job.
+	RunPolicy common.RunPolicy `json:"runPolicy"`
```
Comment (Jeffwan) [inline v2/pkg/apis/kubeflow/v2beta1/types.go:48]: Great. we will have unified interface now. All rest operator have been change to similar pattern.

**Comment by Jeffwan** [comment]
LGTM

File: v2/cmd/mpi-operator/app/server.go [hunk 2]
```diff
@@ -157,6 +157,7 @@ func Run(opt *options.ServerOption) error {
 			kubeInformerFactory.Core().V1().ConfigMaps(),
 			kubeInformerFactory.Core().V1().Secrets(),
 			kubeInformerFactory.Core().V1().Services(),
+			kubeInformerFactory.Batch().V1().Jobs(),
```
Comment (Jeffwan) [inline v2/cmd/mpi-operator/app/server.go:160]: Is this related to current PR?

Comment (gaocegege) [reply v2/cmd/mpi-operator/app/server.go:160]: Job is used for launchers, thus we have to watch it.

Comment (alculquicondor) [reply v2/cmd/mpi-operator/app/server.go:160]: See #391

**Review by terrytangyuan — APPROVED** [review]
LGTM. Would you like to wait for @ahg-g's review?

**Comment by alculquicondor** [comment]
It seems that there are still open questions in #391
